### PR TITLE
Add chapters to episode details screen

### PR DESCRIPTION
**Pull request by MiSikora**
## Description

This PR adds chapters preview to episode details. It also improves couple of other things:
- Data set changes in player container are now applied selectively with appropriate `notifyItemInserted()` and `notifyItemRemoved()`.
- Chapters are now available in player view before playing the episode. They are not extracted beforehand so embedded chapters still require playing but once extracted and saved in the database they will be there before playing.
- I removed snackbar feedback about deselecting last chapter because it is now blocked in the UI.
- `ChaptersViewModel` is now extensively tested.

## Testing Instructions

Test both episode details and player chapters. They should behave almost the same. The only difference is that tapping on an already playing chapter in the episode details doesn't navigate to player. Check if upsell still works.

Test player behavior with tabs. Add to the queue episodes with and without chapters and skip between them. Chapters tab should behave correctly and appear or disappear depending on whether episode has chapters.

## Screenshots or Screencast 

| Light | Dark | Rose | Indigo | Extra dark |
| - | - | - | - | - |
| Screenshot_20240606-120624 | Screenshot_20240606-120650 | Screenshot_20240606-120700 | Screenshot_20240606-120713 | Screenshot_20240606-120727 |

| Dark Contrast | Light Contrast | Electricity | Classic | Radioactivity |
| - | - | - | - | - |
| Screenshot_20240606-120739 | Screenshot_20240606-120750 | Screenshot_20240606-120759 | Screenshot_20240606-120812 | Screenshot_20240606-120832 |

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] ~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~
- [x] ~Any jetpack compose components I added or changed are covered by compose previews~
- [x] ~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~
 
#### I have tested any UI changes...

- [x] with different themes
- [x] ~with a landscape orientation~
- [x] ~with the device set to have a large display and font size~
- [x] ~for accessibility with TalkBack~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>Args</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ashiagr** [comment]
This is looking great! /cc @david-gonzalez-a8c

I'm still reviewing the changes, but can you look at this drag-to-dismiss issue noticed only on the Chapters tab on the full-screen player?

https://github.com/Automattic/pocket-casts-android/assets/1405144/bf748fa9-e189-4108-9cf8-09e9fdd7e55c

**Comment by MiSikora** [comment]
> I'm still reviewing the changes, but can you look at this drag-to-dismiss issue noticed only on the Chapters tab on the full-screen player?

Sure, I'll take a look. Thanks for finding it!

File: modules/services/localization/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -387,7 +387,6 @@ up    <string name="player_sleep_in_one_chapter">Sleeping in 1 chapter</string>
     <!-- %1$d is the number of total chapters. %2$d is the number of hidden chapters. -->
     <string name="number_of_chapters_summary_plural">%1$d chapters &#x2022; %2$d hidden</string>
     <string name="number_of_chapters_summary_singular">1 chapter &#x2022; %d hidden</string>
-    <string name="select_one_chapter_message">Please select at least one chapter</string>
```
Comment (ashiagr) [inline modules/services/localization/src/main/res/values/strings.xml:390]: No action is needed but it is sufficient to just delete the key from the main strings file, keys from other translations will be automatically deleted on next translation sync.

File: modules/features/player/src/test/java/au/com/shiftyjelly/pocketcasts/player/viewmodel/ChaptersViewModelTest.kt [hunk 2]
```diff
@@ -1,168 +1,294 @@
...
+        playbackStateFlow.update { it.copy(state = PlaybackState.State.STOPPED) }
+
+        chaptersViewModel.playChapter(chapters.getList()[2])
+
+        verify(playbackManager, times(1)).skipToChapter(chapters.getList()[2])
+        verifyBlocking(playbackManager, times(1)) { playNowSuspend("id") }
+    }
+
+    @Test
+    fun `play chapter from different episode`() = runTest {
```
Comment (ashiagr) [inline modules/features/player/src/test/java/au/com/shiftyjelly/pocketcasts/player/viewmodel/ChaptersViewModelTest.kt:261]: Love these tests!

**Review by ashiagr — APPROVED** [review]
Tested following:

Chapters tab visibility based on an episode with and without chapters  ✅ 
Chapter selection/ deselection on both episode details and player tab  ✅
Navigation on chapter row tap ✅ 
Upsell triggered for non-paid users  ✅ 
Analytics ✅ 
Theming: Minor issue with checkbox ticks (noticed in Rose theme, but let's check other themes too)

Before | After
---|---
checkmarks | checkmarks_now

**Comment by MiSikora** [comment]
>  Theming: Minor issue with checkbox ticks (noticed in Rose theme, but let's check other themes too)

Fixed in 9d38f66a111d7bd15a5a16d2f603ecc9cf0e6be6

**Pull request merged**